common/queue/src/main/java/org/thingsboard/server/queue/kafka/KafkaAdmin.java

```diff
@@ -0,0 +1,288 @@
+/**
+ * Copyright © 2016-2025 The Thingsboard Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.thingsboard.server.queue.kafka;
+
+import jakarta.annotation.PreDestroy;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.concurrent.ConcurrentException;
+import org.apache.commons.lang3.concurrent.LazyInitializer;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.CreateTopicsResult;
+import org.apache.kafka.clients.admin.ListOffsetsResult;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.admin.OffsetSpec;
+import org.apache.kafka.clients.admin.TopicDescription;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.errors.TopicExistsException;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+import org.thingsboard.common.util.CachedValue;
+import org.thingsboard.server.queue.util.TbKafkaComponent;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+
+@TbKafkaComponent
+@Component
+@Slf4j
+public class KafkaAdmin {
+    /*
+     * TODO: Get rid of per consumer/producer TbKafkaAdmin,
+     *  use single KafkaAdmin instance that accepts topicConfigs.
+     * */
+
+    private final TbKafkaSettings settings;
+
+    private final LazyInitializer<AdminClient> adminClient;
+    private final CachedValue<Set<String>> topics;
+
+    public KafkaAdmin(@Lazy TbKafkaSettings settings) {
+        this.settings = settings;
+        this.adminClient = LazyInitializer.<AdminClient>builder()
+                .setInitializer(() -> AdminClient.create(settings.toAdminProps()))
+                .get();
+        this.topics = new CachedValue<>(() -> {
+            Set<String> topics = ConcurrentHashMap.newKeySet();
+            topics.addAll(listTopics());
+            return topics;
+        }, TimeUnit.MINUTES.toMillis(5));
```

Comment:
please, use @Value with default 5 min. Do not include in yaml. I can always set a value via ENV Spring naming conversion in case of emergency